src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostRawUserInterface.cs

```diff
@@ -1745,10 +1746,10 @@ public override string ToString()
         /// from the keyboard device, blocking processing until a keystroke is
         /// typed that matches the specified keystroke options.
         /// </summary>
-        /// <param name="options">Unused</param>
+        /// <param name="options">Only NoEcho is supported.</param>
         public override KeyInfo ReadKey(ReadKeyOptions options)
         {
-            ConsoleKeyInfo key = Console.ReadKey();
+            ConsoleKeyInfo key = Console.ReadKey(options.HasFlag(ReadKeyOptions.NoEcho));;
```

Comment:
Sorry, I don't see that feedback, so I can't comment on that.

It's not an opinion that the api is significantly slower, e.g. see [here](https://stackoverflow.com/questions/7368652/what-is-it-that-makes-enum-hasflag-so-slow) - that's the second hit for me on this [search](https://www.bing.com/search?q=Enum.HasFlag)

My preference is to avoid the api completely because:
1. We do have places where the api would make PowerShell measurably slower
2. People working in our code base should be comfortable reading and writing code that is equivalent to this api
3. People tend to copy the style, so avoiding the api helps avoid usage creeping into places that matter.

Again, this is just feedback - I'm not blocking the PR.